admin/notifiers/class-subscription-notifier.php

```diff
@@ -0,0 +1,416 @@
+	 */
+	private function apply_utm_tags( $url ) {
+		return WPSEO_Shortlinker::get( $url . $this->get_utm_tags() );
+	}
+
+	/**
+	 * Gets the UTM tags.
+	 *
+	 * @return string The UTM tags.
+	 */
+	private function get_utm_tags() {
+		$utm_tags = array(
+			'utm_source'   => 'yoast-seo',
+			'utm_medium'   => 'software',
+			'utm_content'  => 'renewal-notification',
+			'utm_campaign' => 'wordpress-ad',
+			'utm_term'     => $this->get_utm_term(),
+		);
+
+		return '#' . http_build_query( $utm_tags );
+	}
+
+	/**
+	 * Determines the UTM term that should be output.
+	 *
+	 * @return string The UTM term.
+	 */
+	private function get_utm_term() {
+		switch ( $this->current_notification_id ) {
+			case self::EXPIRATION_WITHIN_4_WEEKS:
+				return '4-weeks-before';
+			case self::EXPIRATION_WITHIN_1_WEEK:
+				return '1-week-before';
+			case self::EXPIRATION_WITHIN_1_DAY:
+				return '1-day-before';
+			case self::EXPIRED_WITHIN_LAST_DAY:
+				return '1-day-after';
+			case self::EXPIRED_MORE_THAN_1_DAY:
+				return '29-days-after';
+		}
+		return '';
+	}
+
+	/**
+	 * Gets the appropriate shortlink based on the expiration date.
+	 *
+	 * @return string The shortlink.
+	 */
+	private function get_shortlink() {
+		switch ( $this->current_notification_id ) {
+			case self::EXPIRATION_WITHIN_4_WEEKS:
+				return WPSEO_Shortlinker::get( 'https://yoa.st/3me' );
+			case self::EXPIRATION_WITHIN_1_WEEK:
+				return WPSEO_Shortlinker::get( 'https://yoa.st/3mf' );
+			case self::EXPIRATION_WITHIN_1_DAY:
+				return WPSEO_Shortlinker::get( 'https://yoa.st/3mg' );
+			case self::EXPIRED_WITHIN_LAST_DAY:
+				return WPSEO_Shortlinker::get( 'https://yoa.st/3mh' );
+			case self::EXPIRED_MORE_THAN_1_DAY:
+				return WPSEO_Shortlinker::get( 'https://yoa.st/3mi' );
+			default:
+				return '';
+		}
+	}
+
+	/**
+	 * Updates the current notification id when needed.
+	 *
+	 * @param string $notification_id Notification id.
+	 *
+	 * @return void
+	 */
+	private function set_current_notification( $notification_id ) {
+		if ( $this->current_notification_id === $notification_id ) {
+			return;
+		}
+
+		$this->current_notification_id = $notification_id;
+		update_option( self::NOTIFICATION_ID, $notification_id );
+		$this->notification_has_changed = true;
+	}
+
+	/**
+	 * Clean up all persistent data related to this class.
+	 *
+	 * @return void
+	 */
+	private function clean_up() {
+		delete_option( self::NOTIFICATION_ID );
+		$this->notification_center->remove_notification_by_id( self::NOTIFICATION_ID );
+	}
+
+	/**
+	 * Gets the first add-on that will expire (or has expired) from the add-on manager.
+	 *
+	 * @return stdClass|null Object representing a subscription.
+	 */
+	protected function determine_subscription_with_earliest_expiry_date() {
+		// Required to make sure we only get the latest data.
+		delete_transient( WPSEO_Addon_Manager::SITE_INFORMATION_TRANSIENT );
+
+		$subscriptions = $this->addon_manager->get_subscriptions_for_active_addons();
```

Comment:
This is the only value you need. I would pass this as a dependency to the constructor.